codegenerator/cli/npm/envio/src/Internal.res

```diff
@@ -0,0 +1,20 @@
+type eventParams
+type eventBlock
+type eventTransaction
+
+@genType
+type genericEvent<'params, 'transaction, 'block> = {
+  params: 'params,
+  chainId: int,
+  srcAddress: Address.t,
+  logIndex: int,
+  transaction: 'transaction,
+  block: 'block,
+}
+
+type event = genericEvent<eventParams, eventTransaction, eventBlock>
+
+external fromGenericEvent: genericEvent<'a, 'b, 'c> => event = "%identity"
+external toGenericEvent: event => genericEvent<'a, 'b, 'c> = "%identity"
+
+type loaderReturn
```

Comment:
Should we maybe scope this into a module called `Event` so that as we extend this file things are name-spaced nicely?

Reply:
I think it's more convenient to have it flat. We don't / shouldn't have name collisions in our system.